### PR TITLE
Fix builds nico

### PR DESCRIPTION
**Pull request by omodei**
This is a big update to fix astromodels build on conda and on the ci environment (Travis).
I used the xspec-modelonly build that sherpa developed (available in the /xspec/dev conda channel). This solved the issues of compiling xspec-modelonly on our own.
Also, there is no need anymore to make the linux conda build inside a docker, travis provide linux machines.
There is some more cleaning to do, after some revision, as we can get rid of the xspec-models only recipe. But or now, merging with the master is needed to have threeml installing the correct astromodels during CI tests.
Also, the python 3.5 builds have been commented out. Once we get the master building we could branch it and fix the python 3 builds.

### CODE REVIEW

File: astromodels/functions/function.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import os
 import re
 from yaml.reader import ReaderError
+from yaml import Loader
```
[the remark this answers is not in the document]
Comment (omodei) [reply astromodels/functions/function.py]: I'l try locally to see what happen if I remove it.

File: astromodels/xspec/factory.py [hunk 2]
```diff
@@ -328,6 +341,18 @@ def get_models(model_dat_path):
 
                 raise ValueError("Illegal identifier name %s" % (par_name))
 
+            if float(default_value) > hard_maximum:
```
[the remark this answers is not in the document]
Comment (omodei) [reply astromodels/xspec/factory.py]: I have set a check. Not sure that was needed, but it doesn't hurt

File: astromodels/functions/function.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (omodei) [reply astromodels/functions/function.py]: Removed. Was unnecessary.

File: astromodels/xspec/factory.py [hunk 3]
```diff
@@ -341,6 +341,10 @@ def get_models(model_dat_path):
 
                 raise ValueError("Illegal identifier name %s" % (par_name))
 
+            if hard_maximum<hard_maximum:
+
+                raise ValueError("Hard maximum (%s) < hard minimum (%s)" %(hard_maximum,hard_maximum))
```
Comment (henrikef) [inline astromodels/xspec/factory.py]: Typo again, should be `%(hard_maximum,hard_minimum)`

Comment (omodei) [reply astromodels/xspec/factory.py]: Yes, of course :-)

**Comment by grburgess** [comment]
All is fine by me.

**Comment by grburgess** [comment]
I'm good with merging if @henrikef agrees

**Comment by henrikef** [comment]
Let's do it! Can you merge? I'm at the airport right now...

**Comment by grburgess** [comment]
You got it! Safe flight!

**Pull request merged**